lib/src/rules/pub/sort_dependencies.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:linter/src/analyzer.dart';
+import 'package:analyzer/src/lint/pub.dart'; // ignore: implementation_imports
+
+const _desc = r'Sort dependencies.';
+
+const _details = r'''
+**DO** sort dependencies in `pubspec.yaml`.
+
+Sorting list of dependencies makes maintenance easier.
+''';
+
+class SortDependencies extends LintRule {
+  SortDependencies()
+      : super(
+            name: 'sort_dependencies',
```

Comment:
"dependencies" is a fairly general term. Perhaps we should name this "sort_pub_dependencies" to be a little more specific.